pkg/test/framework/components/ingress/kube.go

```diff
@@ -42,65 +45,99 @@ var (
 )
 
 type kubeComponent struct {
-	id      resource.ID
-	address string
+	id          resource.ID
+	address     string
+	gatewayType IgType
+	caCert      string
+	tlsCert     string
+	tlsKey      string
 }
 
-func newKube(ctx resource.Context, cfg Config) (Instance, error) {
-	c := &kubeComponent{}
-	c.id = ctx.TrackResource(c)
+// getHTTPAddress returns the ingress gateway address for plain text http requests.
+func getHTTPAddress(env *kube.Environment, cfg Config) (interface{}, bool, error) {
+	// In Minikube, we don't have the ingress gateway. Instead we do a little bit of trickery to to get the Node
+	// port.
```

Comment:
Please make sure this is tested on MiniKube :)

Reply:
I prefer to leave it in a followup PR. I moved existing implementation of getting IP and port into getHTTPAddress. Note that the minikube env is treated differently. I need to figure out what is the difference between MiniKube and GKE when fetching the IP and port for HTTPS.

Reply:
Yeah, ingress differs on different environment. I think it does more benefits to ship this test first on GKE.

Reply:
Sure. Sounds good.